Merging the new Dockerized IRC

danicaharlan opened the pull request:
New and improved Dockerized IRC and webchat :)

Commit:
Merging new and improved Dockerized IRC and webchat

danicaharlan committed:
Removing the /public subdir

It's unnecessary for the functionality of the project, removing it by deleting the placeholder file.

itsrachelfish commented on LICENSE:
```diff
@@ -1,21 +0,0 @@
-MIT License
```
You should probably still have a license on this project... MIT makes sense if we want other companies to use this project. However since these files are pretty specific to the Wetfish IRC network, it might make more sense to use AGPL instead. :woman_shrugging:

danicaharlan replied:
I agree; I've added the AGPL.

itsrachelfish commented on README.md:
```diff
@@ -1,20 +1,29 @@
-To add/update a custom module, place its source code into the ./modules 
-directory, then run './irc refresh modules'. This will restart the server.
+#### Initial setup
+Clone this repo, ``git clone https://github.com/taeganb/ircdocker.git``
```
This step is too specific and shouldn't reference the username or repo on github.

danicaharlan replied:
Yup, I've removed those lines from the readme!

itsrachelfish commented on docker-compose.yml:
```diff
@@ -0,0 +1,33 @@
+  nginx:
+    image: bitnami/nginx
+    volumes:
+      - /home/debian/ircd2/nginx/conf/private/server_block.conf:/opt/bitnami/nginx/conf/server_blocks/my_server_block.conf:ro
```
Doesn't this mean there needs to be a user account called "debian" created on the host machine?

I think it would be better to define the volume with a relative path within the repo

danicaharlan replied:
No - my original idea was to have users of the repo edit those paths to fit their machines (ie /home/ski/repo for me), but I've updated docker-compose with relative paths, I agree, that is a better decision.

itsrachelfish commented on inspircd/conf/public/censor.conf:
```diff
@@ -0,0 +1,6 @@
+<badword text=":dunno:" replace="¯\(º_o)/¯">
+<badword text="pagga" replace="wetfish">
+<badword text="rapechel" replace="poo">
+<badword text="arabs.ps" replace="wetfish.net">
```
these old filters are unnecessary

danicaharlan replied:
Yep! I removed them.

itsrachelfish commented on thelounge/conf/public/config.js:
````diff
@@ -0,0 +1,458 @@
+	//   join: "#thelounge"
+	// }
+	// ```
+	defaults: {
+		name: "Network",
+		host: "your.website.tld",
+		port: 6697,
+		password: "",
+		tls: true,
+		rejectUnauthorized: true,
+		nick: "Anonymous",
+		username: "Anonymous",
+		realname: "Anonymous",
+		join: "#main",
+	},
````
Shouldn't this be `name: "FishNet"`, `host: "irc.wetfish.net"`, and `join: "#wetfish"`?

danicaharlan replied:
It should, and it is now. :)

itsrachelfish reviewed (changes requested):
A few more changes and this will be perfect :+1:

Commit:
A few last changes before merging

danicaharlan commented:
Proper changes have been made! :+1:

danicaharlan commented:
Did not mean to close the pull request! :woman_facepalming:

danicaharlan committed:
Updated the README to match the new repo

danicaharlan committed:
Updated README

Removed a /

itsrachelfish commented on docker-compose.yml:
```diff
@@ -3,8 +3,8 @@ services:
     volumes:
-      - /home/debian/ircd2/nginx/conf/private/server_block.conf:/opt/bitnami/nginx/conf/server_blocks/my_server_block.conf:ro
-      - /home/debian/ircd2/certs:/certs
+      - $HOME/irc.wetfish.net/nginx/conf/private/server_block.conf:/opt/bitnami/nginx/conf/server_blocks/my_server_block.conf:ro
```
It is unnecessary to use `$HOME/irc.wetfish.net`. You should simply use a path relative to the compose file here. 

As per the Docker volume docs: https://docs.docker.com/compose/compose-file/#volumes

```
  # Path on the host, relative to the Compose file
  - ./cache:/tmp/cache
```

danicaharlan replied:
I've made the change to relative paths. Thank you for the doc reference. :+1:

itsrachelfish reviewed (changes requested):
So close I can almost smell the fish

danicaharlan committed:
Updated docker-compose

Now it's using relative paths!

danicaharlan committed:
Updated inspircd.conf

Needed to update a few file paths.

danicaharlan committed:
Updated inspircd.conf

Changed some more file paths. :woman_facepalming:

danicaharlan committed:
Updated inspircd.conf

Added dhfile within the gnutls tag

danicaharlan committed:
Updated modules.conf

Changed the webirc include filepath

danicaharlan committed:
Updated config.js

Renamed thelounge-theme-zenburn to zenburn in order to allow thelounge to load the theme

danicaharlan committed:
Updated Dockerfile

Switched from root to thelounge user for a lot of commands to avoid complications with having things being owned by root.

danicaharlan committed:
Updated Dockerfile

Removed a couple of zeroes in order to avoid having to chown the configuration directory on the host machine. :+1:

danicaharlan committed:
Updated README

Added proper instructions regarding ownership and permissions for certs.

itsrachelfish commented:
The readme should be updated to more accurately reflect the steps that need to be taken in a real server environment. For example, these are the steps I needed to take to just get the project set up to a point where all of the containers were built (without setting up any of the configuration).

```
apt update
apt install docker docker-compose certbot git
certbot certonly
adduser ircd --disabled-password
usermod -aG docker ircd
su ircd
cd
git clone https://github.com/wetfish/irc.wetfish.net
mkdir certs
exit
cp /etc/letsencrypt/live/docker.wetfish.net/* /home/ircd/irc.wetfish.net/certs
chown -Rv ircd:ircd /home/ircd/irc.wetfish.net/certs
chmod 0644 /home/ircd/irc.wetfish.net/certs/*
su ircd
cd ~/irc.wetfish.net/inspircd/conf/private
ln -s /home/ircd/irc.wetfish.net/certs/* ./
cd ~/irc.wetfish.net/thelounge/conf/private
ln -s /home/ircd/irc.wetfish.net/certs/* ./
exit
cd ~/irc.wetfish.net/inspircd
docker build -t inspircd:latest .
cd ~/irc.wetfish.net/thelounge
docker build -t thelounge:latest .
cd ~/irc.wetfish.net
docker-compose up
```

danicaharlan committed:
Updated README with proper, linear instructions; made use of symlinking for less hassle

Commit:
Add bootstrap script and fix UID issues

danicaharlan committed:
Update README

danicaharlan committed:
Update README

danicaharlan committed:
Update Dockerfile

danicaharlan committed:
Update Dockerfile

danicaharlan committed:
Fix oper status granting in unregistered channels

Yam-s committed:
Use docker bridge ip for inspircd webirc mask

Yam-s committed:
Add permchannels.conf to inspircd config folder

Yam-s committed:
Use pre-built inspircd and thelounge images and fix thelounge config volume

Yam-s committed:
Fix ip range for inspircd webirc

Yam-s committed:
Re-write fishboot.sh to use pre-built docker images

Yam-s committed:
Add HTTPs redirect to nginx example config

Yam-s committed:
Rename permchannels.conf to permchannels.conf.example

Yam-s committed:
Update .gitignore to ignore secrets

Yam-s committed:
Force certificate renewal

Yam-s committed:
Rename fishboot.sh -> install.sh

Yam-s committed:
Prevent copying Dockerfiles to install location

Yam-s committed:
Don't use unnecessary temp folder to store certificates

Yam-s committed:
Add config.example.sh

Yam-s committed:
Add cronjob for renewing certificates

Yam-s committed:
Use absolute path for cronjob

Yam-s committed:
Update docker-compose repositories

Yam-s committed:
Add initial support for multiple domains

Yam-s committed:
Remove unnecessary directory removal

Yam-s committed:
Use $PWD instead of absolute path for cronjob

Yam-s committed:
Remove staging flag

Yam-s committed:
Update installation instructions

Yam-s committed:
Simplify configuration checks

Yam-s committed:
Use /etc/cron.d/ rather than modifying crontab directly

Yam-s committed:
Fix nginx reverse proxy and use docker hostnames

Yam-s committed:
Restructure inspircd config directories

Yam-s committed:
Reload certificates within post-deploy cronjob

Yam-s committed:
Update Dockerfiles

Yam-s committed:
Make adduser non-interactive

Yam-s committed:
Remove check for existing user

itsrachelfish reviewed (approved):
Set up a new production server using this branch. Was able to link the dockerized server to the network
